e2e/pageobjects/ide/Terminal.ts

```diff
@@ -0,0 +1,116 @@
+import { CLASSES } from '../../inversify.types';
+import { DriverHelper } from '../../utils/DriverHelper';
+import { By, Key, WebElement, error } from 'selenium-webdriver';
+import { TestConstants } from '../../TestConstants';
+
+@injectable()
+export class Terminal {
+    constructor(@inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) { }
+
+    async waitTab(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const terminalTabLocator: By = By.css(this.getTerminalTabCssLocator(tabTitle));
+
+        await this.driverHelper.waitVisibility(terminalTabLocator, timeout);
+    }
+
+    async waitTabAbsence(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const terminalTabLocator: By = By.css(this.getTerminalTabCssLocator(tabTitle));
+
+        await this.driverHelper.waitDisappearanceWithTimeout(terminalTabLocator, timeout);
+    }
+
+    async clickOnTab(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const terminalTabLocator: By = By.css(this.getTerminalTabCssLocator(tabTitle));
+
+        await this, this.driverHelper.waitAndClick(terminalTabLocator, timeout);
+    }
+
+    async waitTabFocused(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const focusedTerminalTabLocator: By = this.getFocusedTerminalTabLocator(tabTitle);
+
+        await this.driverHelper.waitVisibility(focusedTerminalTabLocator, timeout);
+    }
+
+    async selectTerminalTab(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        await this.clickOnTab(tabTitle, timeout);
+        await this.waitTabFocused(tabTitle, timeout);
+    }
+
+    async clickOnTabCloseIcon(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const terminalTabCloseIconLocator: By =
+            By.css(`${this.getTerminalTabCssLocator(tabTitle)} div.p-TabBar-tabCloseIcon`);
+
+        await this.driverHelper.waitAndClick(terminalTabCloseIconLocator, timeout);
+    }
+
+    async closeTerminalTab(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        await this.clickOnTabCloseIcon(tabTitle, timeout);
+        await this.waitTabAbsence(tabTitle, timeout);
+    }
+
+    async type(terminalTabTitle: string, text: string) {
+        const terminalIndex: number = await this.getTerminalIndex(terminalTabTitle);
+        const terminalInteractionContainer: By = this.getTerminalEditorInteractionEditorLocator(terminalIndex);
+
+        await this.driverHelper.typeToInvisible(terminalInteractionContainer, text);
+    }
+
+    async rejectTerminalProcess(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        await this.selectTerminalTab(tabTitle, timeout);
+
+        // for ensure that element is interactable
+        await this.driverHelper.wait(3000);
```

Comment:
Again... couldn't we rewrite these sleeps/waits to be "active" wait? What does it mean for the element to be interactable? Couldn't we write this down as a code which waits for this?

Reply:
I guess this is still something I strongly feel about... Until I'm proven we **really need** this and it couldn't be done any other way I would request this to change.